streambed-logged/src/lib.rs

```diff
@@ -262,18 +272,42 @@ impl CommitLog for FileLog {
                 );
                 drop(locked_topic_file_ops);
 
-                let mut next_offset = find_offset(&topic_file_op, self.read_buffer_size, false)
-                    .ok()
-                    .flatten()
-                    .map_or(0, |offsets| offsets.end_offset.wrapping_add(1));
+                let found_offsets = match find_offset(
+                    &topic_file_op,
+                    self.read_buffer_size,
+                    self.max_record_size,
+                    false,
+                ) {
+                    r @ Ok(_) => r,
+                    Err(e) => {
+                        error!("Error {e} when producing. Attempting to recover by truncating the active file.");
+
+                        if let Err(e) = recover_active_file(
+                            &mut topic_file_op,
+                            self.read_buffer_size,
+                            self.max_record_size,
+                        ) {
+                            error!("Error {e} when recoverying. Unable to recover the active file.")
+                        }
+
+                        find_offset(
+                            &topic_file_op,
+                            self.read_buffer_size,
+                            self.max_record_size,
+                            false,
+                        )
+                    }
+                };
+                let mut next_offset = found_offsets
+                    .map(|offsets| offsets.map_or(0, |offsets| offsets.end_offset.wrapping_add(1)));
```

Comment:
Factored out the recovery logic into its own logic. The changes here highlight that there's little impact to the code b base overall.